### PR TITLE
Ik 79

### PR DESCRIPTION
**Pull request by handriamanantena**
This branch contains the changes which will check if the user has logged in for the first time by calling a script and if the user is logged in for the first time the user will be brought to a gender/smoking page.

### CODE REVIEW

File: Backend/getPassengersOnRoute.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+require "init.php";
+
+$user_email= "anosirrah365@gmail.com";//$_POST["loggedInUserEmail"];
+$routeName= "From Deez To Nutz";//$_POST["routeName"];
```
Comment (adrielFab) [inline Backend/getPassengersOnRoute.php]: getPassengersOnRoute.php is still hardcoded. Update the file

File: client/app/src/main/java/Route/RouteView.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Map<String, String> jsonbody = new HashMap<>();
+        Map<String, String> userInfo = UserSerializer.getParameters(RequestHandler.getUser());
+        requestHandler.httpPostStringRequest("http://"+context.getString(R.string.web_server_ip)  +
+                "/getPassengersOnRoute.php",jsonbody,
+                RequestHandler.URLENCODED ,context);
+
+    }
+
+    @Override
+    public void Update(String response) {
```
Comment (adrielFab) [inline client/app/src/main/java/Route/RouteView.java]: Empty body method

File: Backend/getPassengersOnRoute.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+//get route id
+$sql = "SELECT route_id FROM Routes_Users_Association WHERE user_id  = (SELECT id FROM Users WHERE email = '".$user_email."')
+            AND userRouteName = '".$routeName."'";
+    
+$result = mysqli_query($con,$sql);
+
+
+if(mysqli_num_rows($result) == 0){
```
Comment (SaifMahabub) [inline Backend/getPassengersOnRoute.php]: Control statement "if" should have one space between the control keyword and opening parenthesis according to PHP conventions.

File: client/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -63,7 +63,11 @@
...
         </activity>
-        <activity android:name=".Activity.RouteActivity"
+        <activity
+            android:name=".Activity.RouteActivity"
+            android:theme="@style/AppTheme2">
+        </activity>
+        <activity android:name=".Activity.FirstTimeActivity"
             android:theme="@style/AppTheme2">
         </activity>
     </application>
```
Comment (SaifMahabub) [inline client/app/src/main/AndroidManifest.xml:73]: Control statements "if", "while" and "else if" should have one space between the control keyword and opening parenthesis according to PHP conventions.

File: client/app/src/main/java/Route/RouteView.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.Map;
+
+public class RouteView implements ActivityObserver{
+
+    private Context context;
+    private List<Passenger> listOfPassengers = new ArrayList<>();
+    private LinearLayout.LayoutParams layout;
+    private RequestHandler requestHandler = new RequestHandler();
+
+    public RouteView(LinearLayout.LayoutParams ll, final Context context){
```
Comment (SaifMahabub) [inline client/app/src/main/java/Route/RouteView.java]: Missing space before opening curly brackets.

File: client/app/src/main/java/Route/RouteView.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        this.layout = ll;
+        this.context = context;
+    }
+
+    public void displayButtons(String buttons){
```
Comment (SaifMahabub) [inline client/app/src/main/java/Route/RouteView.java]: Missing space before opening curly brackets.

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 7]
```diff
@@ -164,15 +164,22 @@ private void firebaseAuthWithGoogle(GoogleSignInAccount acct) {
      */
     @Override
     public void Update(String response) {
-
+        mProgressDialog.dismiss();
         requestHandler.detach(this);
-        System.out.println("Response: "+ response);
-        if(response.contains("User")) {
+        if(response.contains("True")){ // this is not the first time the user is logged in
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: Missing space before opening curly brackets.

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 8]
```diff
@@ -164,15 +164,22 @@ private void firebaseAuthWithGoogle(GoogleSignInAccount acct) {
...
+            Intent intent = new Intent(MainActivity.this, HomePage.class);
+            this.startActivity(intent);
+        }
+        else if(response.contains("False")){ // this is the first time the user is logged in
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: Missing space before opening curly brackets.

File: client/app/src/main/java/com/example/mrides/login/LoginFirstTime.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import com.example.mrides.userDomain.UserSerializer;
+
+public class LoginFirstTime implements ActivityObserver{
+
+    private FirstTimeActivity activity;
+    private RequestHandler requestHandler = new RequestHandler();
+    private ProgressDialog mProgressDialog;
+
+
+    public LoginFirstTime(FirstTimeActivity activity){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/login/LoginFirstTime.java:23]: Missing space before opening curly brackets.

**Review by SaifMahabub — CHANGES_REQUESTED** [review]

**Review by SaifMahabub — APPROVED** [review]

**Pull request merged**